syntax = "proto3";

service CalculationService {
  rpc PerformTotalCalculation (CalculationRequest) returns (CalculationResponse);
}

service OperationService {
  rpc PerformPartialCalculation (PartialCalculationRequest) returns (PartialCalculationResponse);
}

message CalculationRequest {
  double operand1 = 1;
  double operand2 = 2;
}

message CalculationResponse {
  double result = 1;
}

message PartialCalculationRequest {
  double value = 1;
}

message PartialCalculationResponse {
  double partial_result = 1;
}
